NTT - Get Started page (#376)
wormhole-foundation/wormhole-docs: products/native-token-transfers/get-started.md

---

> title: Get Started

Reviewer: Please make this more descriptive for SEO purposes. If this was to show up in search engine results, it's like "ok, get started with what? wormhole as a whole?"

> description: Set up your environment, install the NTT CLI, and scaffold your first cross-chain token configuration for deployment.

Reviewer: Only 117 characters. Needs to be a minimum of 120. This also provides no context into what NTT is. Might be helpful to add more context for those coming across this as a result in search engines

categories: NTT, Transfer
---

> # Get Started

Reviewer 2: ```suggestion
# Get Started with NTT
```

Reviewer 2: I'll defer to @eshaben  here, but I like having the product name in this title.

Reviewer: Sounds good to me

## Introduction

> Native Token Transfers (NTT) enable seamless cross-chain token movement without wrapping or liquidity pools. This guide will help you install the NTT CLI, scaffold your first NTT configuration, and prepare for deployment.

Reviewer: I think we're talking about the framework here rather than transfers themselves. So should be "enables"

Reviewer: Why do they need the NTT CLI?

Reviewer: For the first reference of Native Token Transfers (NTT), I'd like to see a link back to the overview section so users can learn more if they want

> Native Token Transfers (NTT) enable seamless cross-chain token movement without wrapping or liquidity pools. This guide will help you install the NTT CLI, scaffold your first NTT configuration, and prepare for deployment.

Reviewer 3: there should be a link here to overview or concepts

Author: There is, weird... you might be looking at a previous version

Reviewer 3: haha idk man i just fetched the branch

If you are looking for a no-code experience to deploy on mainnet, you can explore the [NTT Launchpad](https://ntt.wormhole.com){target=\_blank}.

## Prerequisites

Before you begin, make sure you have:

- Node.js and npm installed
- Bun installed
- A wallet private key with tokens on supported chains
- ERC-20 or SPL tokens already deployed on the source and destination chains

## Don’t Have a Token Yet?

To use NTT, you must have a token already deployed on the source and destination chains. If you don’t have one, follow one of the quick guides below to deploy a basic test token.

> ???- interface "Deploy an ERC-20 Token on EVM"

Reviewer 2: I've made some notes on how we might improve how we share this info in the future phase but nothing that would be a blocker to putting this version out.

> To use NTT, you must have a token already deployed on the source and destination chains. If you don’t have one, follow one of the quick guides below to deploy a basic test token.

Reviewer 3: 'one' is being repeated, i would remove one

        ```bash
        cargo install spl-token-cli
        ```

    6. **Create a new SPL token** - initialize the token on Solana

        ```bash
        spl-token create-token
        ```

    7. **Create a token account** - generate an account to hold the token

        ```bash
        spl-token create-account INSERT_TOKEN_ADDRESS
        ```

    8. **Mint tokens** - send 1000 tokens to the created account

        ```bash
        spl-token mint INSERT_TOKEN_ADDRESS 1000
        ```

    !!! note
        NTT versions `>=v2.0.0+solana` support SPL tokens with [transfer hooks](https://spl.solana.com/transfer-hook-interface){target=\_blank}.

## Install NTT CLI

The NTT CLI is recommended to deploy and manage your cross-chain token configuration.

1. Run the installation command in your terminal:

    ```bash
    curl -fsSL https://raw.githubusercontent.com/wormhole-foundation/native-token-transfers/main/cli/install.sh | bash
    ```

2. Verify the NTT CLI is installed:

    ```bash
    ntt --version
    ```

## Update NTT CLI

To update an existing NTT CLI installation, run the following command in your terminal:

```bash
ntt update

> ```

Reviewer: Could probably move to this the NTT CLI page instead. Doesn't seem necessary for a Get Started guide because if they're following along, they already just installed the latest. Wdyt?

Author: I agree. I added it thinking about the case where a user had it installed in the past, but it makes sense that it shouldn't be here as a "Get Started" guide.

## Initialize a New NTT Project

Once the CLI is installed, scaffold a new project by running:

```bash
ntt new my-ntt-project
cd my-ntt-project
```

Then, initialize a new `deployment.json` file specifying the network:

=== "Mainnet"
    ```bash
    ntt init Mainnet
    ```

=== "Testnet"
    ```bash
    ntt init Testnet
    ```

> This file contains your NTT configuration and starts with the selected network after initialization. 

Reviewer: Please convert these to an ordered list

> This file contains your NTT configuration and starts with the selected network after initialization. 

Reviewer 3: ```suggestion
The `deployment.json` file contains your NTT configuration and starts with the selected network after initialization. 
```

???- code "Example `deployment.json` for testnet"
    ```json

>     {
>     "network": "Testnet",
>     "chains": {}
>     }

Reviewer: ```suggestion
    {
        "network": "Testnet",
        "chains": {}
    }
```

4 spaces for JSOIN

Reviewer: Wdyt about switching this to tabs and adding one for mainnet? So it follows format of previous step. Also, since this is so small we could move this out of an expandable element.

    ```

> In the [deployment steps](#next-steps), you will add your supported chains, along with their token addresses, deployment modes, and any custom settings.

Reviewer 2: Not sure we need the link since the item it links to is directly underneath. It doesn't move when you click it which makes it seem like a broken link.
